### ISSUE TITLE
[BUG] In-Game Config not saving

### ISSUE THREAD

**Issue by MarkinhosMLP** [trainb0y/FabriZoom#4]
ive been changing mouse sensitivity and it always come back at 40 the same happens with minimum and max zoom divisor

**Comment by trainb0y** [comment]
Can you send your Minecraft log? The mod logs every time it saves or loads the config, so it might be helpful because I can't reproduce this. 

[mclo.gs](https://mclo.gs/) is good place to upload and it automatically removes personal data.

**Comment by MarkinhosMLP** [comment]
https://mclo.gs/HaXZwXK

**Comment by MarkinhosMLP** [comment]
Uploading fabriczoomerbug.mp4…

**Comment by trainb0y** [comment]
Looks like you’re on an outdated version (v1.1.0). v1.1.1 should hopefully have fixed it.

**Comment by MarkinhosMLP** [comment]
> Looks like you’re on an outdated version (v1.1.0). v1.1.1 should hopefully have fixed it, assuming it’s the issue I think it is.

1.1.1 have no way to access the configuration iirc, let me try again

**Comment by MarkinhosMLP** [comment]
tried again in 1.1.1 and stills the same bug...

**Comment by trainb0y** [comment]
It won’t save, or the config won’t open?

**Comment by MarkinhosMLP** [comment]
> It won’t save, or the config won’t open?

i reinstalled mod menu and the config opened normally, but it stay without saving the config i set

**Comment by trainb0y** [comment]
Can you send another log?

**Comment by MarkinhosMLP** [comment]
https://mclo.gs/AwaGqND

**Comment by MarkinhosMLP** [comment]
https://user-images.githubusercontent.com/47507632/195475992-8fc0f177-5350-4de8-940e-c69bc841a7a2.mp4

**Comment by trainb0y** [comment]
> https://mclo.gs/AwaGqND

Can you send the log from when the bug happens? This log doesn't have anything from FabriZoom as far as I can tell.

**Comment by MarkinhosMLP** [comment]
> > https://mclo.gs/AwaGqND
> 
> Can you send the log from when the bug happens? This log doesn't have anything from FabriZoom as far as I can tell.
> 
> Very confused because I can't get this to happen myself.

that was the log from when it happened
i restarted the game and just joined a world and used the zoom and changed the config, then, i quit the game . the bug keeps happening, heres the log https://mclo.gs/I5hsdTr

**Comment by trainb0y** [comment]
Well, to be honest I'm kinda at a loss... it might be a mod conflict, but I doubt it. 
According to the log, everything should be fine. (At least fabrizoom-wise, looks like optifine is having a stroke)

One last idea, can you send `.minecraft/config/fabrizoom.conf`?

**Comment by MarkinhosMLP** [comment]
[fabrizoom.conf.txt](https://github.com/trainb0y/FabriZoom/files/9769838/fabrizoom.conf.txt)

sending it as a txt cause github doesnt supports .conf files

**Comment by trainb0y** [comment]
Writing this out, not because I think you need to know it, but because writing it out might help me think through this.
<details>
<summary>Config States</summary>

   #### Saves to file when  
   - Config screen exited  
   - Save button pressed  
   - Reset button pressed
   #### Loads from file when  
   - Initial mod load
  - Cancel button pressed

SpruceUI (the GUI lib the config screen uses) doesn't update the config fields except when the screen is first created, as far as I can tell. 

</details>

My best guess is that whatever mod you are using for the shortcut to the menu doesn't create a new screen like it should, and doesn't actually close it when you exit it, and therefore doesn't save the config. However, if that was the case the screen should still be displaying whatever value you set, not the default of 40. I could be completely wrong, idk. 

Here are the last two things I can think of to try, if you want to:
1. Manually editing `fabrizoom.conf` before starting the game to see if it keeps the values or gets overwritten
1. manually pressing the save button to exit the config screen.

If neither of those work, I'm out of ideas.

**Comment by trainb0y** [comment]
OH oh I'm an idiot, ignore that. :facepalm: 

Mouse sensitivity is automatically clamped between 10 and 40. It shouldn't ever end up displaying 80 in the config screen, which is a bug, but a much more minor one.

**Comment by trainb0y** [comment]
Mouse sensitivity can go up to 60 as of https://github.com/trainb0y/FabriZoom/commit/1fda8a92b84cfc185be79fe211b285bd2fa455b2

To fix the actual underlying bug is going to take more effort, it might be worth migrating to a different GUI library, as SpruceUI doesn't provide an easy way to force `SpruceOption`s to update their contents.

**Comment by trainb0y** [comment]
Alright I’m going to close this issue because the actual bug isn’t related to the config saving and loading, see #5
